component-samples/component-http-client-to0-sample/src/main/java/org/fido/iot/sample/To0ClientComponentApp.java

```diff
@@ -0,0 +1,310 @@
+        if (bitLength == Const.BIT_LEN_256) {
+          pemValue = sampleOwnerKeyPemEC256;
+        } else if (bitLength == Const.BIT_LEN_384) {
+          pemValue = sampleOwnerKeyPemEC384;
+        }
+      } else if (ownerPublicKey instanceof RSAKey) {
+        pemValue = sampleOwnerKeyPemRSA;
+      }
+      return PemLoader.loadPrivateKey(pemValue);
+    }
+
+    @Override
+    public void starting(Composite request, Composite reply) {
+
+    }
+
+    @Override
+    public void started(Composite request, Composite reply) {
+
+    }
+
+    @Override
+    public void continuing(Composite request, Composite reply) {
+      Composite info = request.getAsComposite(Const.SM_PROTOCOL_INFO);
+      if (info.containsKey(Const.PI_TOKEN)) {
+        clientToken = info.getAsString(Const.PI_TOKEN);
+      }
+      reply.set(Const.SM_PROTOCOL_INFO,
+          Composite.newMap().set(Const.PI_TOKEN, clientToken));
+    }
+
+    @Override
+    public void continued(Composite request, Composite reply) {
+
+    }
+
+    @Override
+    public void completed(Composite request, Composite reply) {
+
+    }
+
+    @Override
+    public void failed(Composite request, Composite reply) {
+
+    }
+  };
+
+  To0ClientService clientService = new To0ClientService() {
+
+    @Override
+    protected To0ClientStorage getStorage() {
+      return clientStorage;
+    }
+
+    @Override
+    public CryptoService getCryptoService() {
+      return cryptoService;
+    }
+  };
+
+  private MessageDispatcher createDispatcher() {
+
+    return new MessageDispatcher() {
+
+      @Override
+      protected MessagingService getMessagingService(Composite request) {
+        return clientService;
+      }
+
+      @Override
+      protected void failed(Exception e) {
+        System.out.println(e.getMessage());
+      }
+    };
+  }
+
+  private void run(String[] args)
+      throws NoSuchAlgorithmException, IOException, InterruptedException {
+
+    cryptoService = new CryptoService();
+
+    MessageDispatcher dispatcher = createDispatcher();
+
+    DispatchResult dr = clientService.getHelloMessage();
+
+    Composite ovh = clientStorage.getVoucher().getAsComposite(Const.OV_HEADER);
+    Composite rvi = ovh.getAsComposite(Const.OVH_RENDEZVOUS_INFO);
+
+    List<String> paths = RendezvousInfoDecoder.getHttpDirectives(
+        rvi, Const.RV_OWNER_ONLY);
+
+    for (String path : paths) {
+
+      responseWait = null;
+      try {
+        WebClient client = new WebClient(path, dr, dispatcher);
+        client.run();
+        if (responseWait != null) {
+          break;
+        }
+      } catch (Exception e) {
+        System.out.println(e.getMessage());
+      }
+    }
+
+  }
+
+  /**
+   * Application main.
+   *
+   * @param args The application arguments.
+   */
+  public static void main(String[] args)
+      throws NoSuchAlgorithmException, IOException, InterruptedException {
+    new To0ClientComponentApp().run(args);
+    System.out.println("TO0 Client finished.");
```

Comment:
Consider extracting GUID from voucher and printing it along with sout. [Good to have]